source/Nuke.Common/Constants.cs

```diff
@@ -34,7 +34,18 @@ internal static class Constants
         internal const string GlobalToolStartTimeEnvironmentKey = "NUKE_GLOBAL_TOOL_START_TIME";
 
         internal static AbsolutePath GlobalTemporaryDirectory => (AbsolutePath) Path.GetTempPath();
-        internal static AbsolutePath GlobalNukeDirectory => (AbsolutePath) Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) / ".nuke";
+
+        [CanBeNull]
+        internal static AbsolutePath GlobalNukeDirectory
+        {
+            get
+            {
+                var folderPath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                if (string.IsNullOrEmpty(folderPath))
```

Comment:
Running
```
docker run -it --rm <image> pwsh
echo [System.Environment]::GetFolderPath([System.Environment+SpecialFolder]::UserProfile)
```

returns:
| image | windows | linux |
| ----------| -------- | ----- | 
| `mcr.microsoft.com/dotnet/sdk:5.0` | `null` | `/root` |
| `mcr.microsoft.com/dotnet/sdk:6.0` | `null` | `/root` |


```
docker run -it --rm <image> pwsh 
echo [System.Environment]::GetEnvironmentVariable("USERPROFILE")
```

returns:
| image | windows | linux |
| ----------| -------- | ----- | 
| `mcr.microsoft.com/dotnet/sdk:5.0` | `C:\Users\ContainerUsers` | `null` |
| `mcr.microsoft.com/dotnet/sdk:6.0` | `C:\Users\ContainerUsers` | `null` |

Reply:
doesn't this look like a bug in the image?

Reply:
Possibly... But how long would that take to fix??
(I'm kinda looking at it as "sometimes you've gotta deal with the world as it is, rather than as you'd like it to be")

Reply:
wasn't meant as "won't fix" 😀 just making sure I'm on the same page

Reply:
:smile: all good

I kinda looked at it (earlier this morning) as a "I should go search on the github and find if this is a known issue", but then went with the "we just handle it anyway"